executor/joiner.go

```diff
@@ -142,6 +142,16 @@ func (j *baseJoiner) makeJoinRowToChunk(chk *chunk.Chunk, lhs, rhs chunk.Row) {
 	chk.AppendPartialRow(lhs.Len(), rhs)
 }
 
+// makeShadowJoinRow combines inner, outer row into one join row and append to the chunk.
+// The result of join row will always append to the dst chunk's first row.
+func makeShadowJoinRow(isRight bool, inner, outer chunk.Row, mutRow chunk.MutRow) {
+	if !isRight {
```

Comment:
It is worth to comment why we switch the order of inner and outer join table when `isRight` being false.